core/src/main/java/org/elasticsearch/common/settings/Setting.java

```diff
@@ -591,6 +587,18 @@ public static ByteSizeValue parseByteSize(String s, ByteSizeValue minValue, Byte
         return value;
     }
 
+    public static Setting<ByteSizeValue> memorySizeSetting(String key, ByteSizeValue value, Property... properties) {
```

Comment:
I'd also appreciate some tests here for these methods and also for the settings that are supposed to accept percentages